packages/core/src/common/cluster-store/cluster-store.ts

```diff
@@ -64,7 +62,7 @@ export class ClusterStore extends BaseStore<ClusterStoreModel> {
 
     const cluster = clusterOrModel instanceof Cluster
       ? clusterOrModel
-      : this.dependencies.createCluster(
+      : new Cluster(
```

Comment:
It's not good thing to remove the `createCluster` since it abstracts the fact that Cluster is even a class. `new` keyword should never be used in the implementation code. There should always be a factory function to abstract the constructor arguments. This allows us to change the implementation how ever we want, but using `new Cluster` strictly ties us down to `Cluster` class with specific arguments.